pkg/apis/core/types.go

```diff
@@ -1603,6 +1606,37 @@ type CSIPersistentVolumeSource struct {
 	NodePublishSecretRef *SecretReference
 }
 
+// Represents a source location of a volume to mount, managed by an external CSI driver
+type CSIVolumeSource struct {
+	// Driver is the name of the driver to use for this volume.
+	// Required.
+	Driver string
+
+	// Optional: The value to pass to ControllerPublishVolumeRequest.
+	// Defaults to false (read/write).
+	// +optional
+	ReadOnly *bool
+
+	// Filesystem type to mount. Ex. "ext4", "xfs", "ntfs".
+	// If not provided, the empty value is passed to the associated CSI driver
+	// which will determine the default filesystem to apply.
+	// +optional
+	FSType *string
+
+	// VolumeAttributes store immutable properties of the volume copied during provision.
+	// These attributes are passed back to the driver during controller publish calls.
```

Comment:
This comment is confused. Inline volume requests can only participate in mount/unmount volume operations, so I guess these attributes will be passed back to the driver during node publish calls, not controller publish calls.

Reply:
Good catch, I will update. You are right it is for NodePublish.